src/rust/engine/process_execution/src/lib.rs

```diff
@@ -431,8 +431,8 @@ pub enum ProcessExecutionStrategy {
     Local,
     /// Stores the platform_properties.
     RemoteExecution(Vec<(String, String)>),
-    /// Stores the image name.
-    Docker(String),
+    /// Stores the image name and mount info.
+    Docker(String, Option<Vec<String>>),
```

Comment:
I recommend switching this to be a struct variant instead of a tuple variant because otherwise readers may become confused with having to remember the order of the values versus just using a field name.

Als, I imagine we are likely to add more fields in the future and a struct will prevent having to keep adding `_` wildcards everywhere to match the tuple form. (The wildcard pattern match then becomes `ProcessExecutionStrategy::Docker(..)` no matter how many fields.)

And then you